tests/test_covariance_real_base.py

```diff
@@ -185,28 +208,34 @@ def _get_fourier_block(self, tracer_comb1, tracer_comb2):
         ("src0", "src0"),
     ],
 )
-def test__get_covariance_block_for_sacc(tracer_comb1, tracer_comb2):
-    cpr = CovarianceProjectedRealTester(input_yml_real)
-    lmax = cpr.lmax
+def test__get_covariance_block_for_sacc(
+    cov_prj_real, tracer_comb1, tracer_comb2
+):
+    lmax = cov_prj_real.lmax
     fourier_block = np.random.rand(lmax + 1, lmax + 1)
 
-    class CovTester(CovarianceProjectedRealTester):
-        def _get_fourier_block(self, tracer_comb1, tracer_comb2):
-            if tracer_comb1 == tracer_comb2:
-                return fourier_block + fourier_block.T
-            return fourier_block
+    # Dynamically override the method on this instance.
```

Comment:
Can you explain a bit what is happening here?

Reply:
What magic functools does?

Reply:
You need to pass extra information like type to do this in plain python, partial wraps it up and makes it easier (https://stackoverflow.com/a/54662690):

```
class MyClass():

    def my_func():
        return "Foo"

my_class = MyClass()

def my_func2():
    return "Bar"

func_type = type(MyClass.my_func)
# Override function
my_class.my_func = func_type(my_func2, my_func, MyClass)

my_class.my_func()
# Prints "Bar"
```